### PR TITLE
fix: improve docs + modify copy of template

### PR DESCRIPTION
**Pull request by digimaun**
* Doc improvements for readme, verify-host example and clearer error for RSR permission check.
* +Permission check unit test content and addition of one more valid form of permission `*/write`.
*  WorkManager.build_template() gets and modifies deep copy of base template.
* Increment version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ on.
 ## Pre-requisites
 
 - An Azure Arc-enabled CNCF K8s cluster.
-- Azure CLI `2.42.0` or higher. If you don't have az cli installed, follow [these instructions](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli).
+- Azure CLI `2.46.0` or higher. If you don't have az cli installed, follow [these instructions](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli).
```
Comment (vilit1) [inline README.md:12]: is there a reason for this?

File: README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ on.
 The IoT Ops CLI is part of the Azure CLI extension index. Installation is as simple as:
 
 ```
-az extension add --name azure-iot-ops
+az extension add --upgrade --name azure-iot-ops
```
Comment (vilit1) [inline README.md:23]: does this work with the b versions?

**Review by c-ryan-k — APPROVED** [review]
👍

File: README.md (hunk 1 above)
Comment (c-ryan-k) [reply README.md:12]: @vilit1 this is our [current min version](https://github.com/Azure/azure-iot-ops-cli-extension/blob/ffe03041b21b51b2ab1be5b8db287a31a3e65fb3/azext_edge/azext_metadata.json#L2)

Comment (vilit1) [reply README.md:12]: as in when did we bump the min version

Comment (digimaun) [reply README.md:12]: To add-on, we need at least 2.46 to ensure track 2 authorization client from CLI core.

File: README.md (hunk 2 above)
Comment (digimaun) [reply README.md:23]: It should
image

File: azext_edge/tests/edge/init/test_permissions_unit.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+from azext_edge.edge.providers.orchestration.permissions import verify_write_permission_against_rg
+from azure.cli.core.azclierror import ValidationError
+
+from ...generators import get_zeroed_subscription, generate_generic_id
+
+MOCK_SUBSCRIPTION_ID = get_zeroed_subscription()
+MOCK_RG = f"rg_{generate_generic_id()}"
+
+
+class IntermObj:
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_permissions_unit.py:18]: assuming this is to stop having .return_value.as_dict.return_value -> we could have this as a general object for tests

**Review by vilit1 — APPROVED** [review]

File: azext_edge/tests/edge/init/test_permissions_unit.py (hunk 3 above)
Comment (digimaun) [reply azext_edge/tests/edge/init/test_permissions_unit.py:18]: yea , works for me

**Pull request merged**